pupil_src/shared_modules/player_methods.py

```diff
@@ -194,23 +195,124 @@ def correlate_data(data, timestamps):
 
 
 def load_meta_info(rec_dir):
-    meta_info_path = os.path.join(rec_dir, "info.csv")
-    with open(meta_info_path, "r", encoding="utf-8") as csvfile:
-        meta_info = csv_utils.read_key_value_file(csvfile)
-    return meta_info
+    return Pupil_Recording(rec_dir).meta_info
 
 
 def is_pupil_rec_dir(rec_dir):
-    if not os.path.isdir(rec_dir):
-        logger.error("No valid dir supplied ({})".format(rec_dir))
-        return False
     try:
-        meta_info = load_meta_info(rec_dir)
-        meta_info["Recording Name"]  # Test key existence
-    except:
-        logger.error("Could not read info.csv file: Not a valid Pupil recording.")
+        Pupil_Recording(rec_dir)
+        return True
+    except InvalidRecordingException as e:
+        logger.error(str(e))
         return False
-    return True
+
+
+class InvalidRecordingException(Exception):
+    def __init__(self, reason: str, recovery: str = ""):
+        message = reason + "\n" + recovery if recovery else reason
+        super().__init__(message)
+        self.reason = reason
+        self.recovery = recovery
+
+    def __str__(self):
+        return f"{type(self).__name__}: {super().__str__()}"
+
+
+class Pupil_Recording:
+    def __init__(self, rec_dir):
+        self._rec_dir = rec_dir
+        self._meta_info = None
+        self.load(rec_dir=rec_dir)
+
+    @property
+    def rec_dir(self):
+        return self._rec_dir
+
+    @property
+    def meta_info(self):
+        return self._meta_info
+
+    def reload(self):
+        self.load(rec_dir=self.rec_dir)
+
+    def load(self, rec_dir):
+
+        def is_video_file(file_path):
+            if not os.path.isfile(file_path):
+                return False
+            # TODO: Are there any other valid extensions?
+            valid_video_extensions = [".mp4"]
+            _, ext = os.path.splitext(file_path)
+            if ext not in valid_video_extensions:
+                return False
+            return True
+
+        if not os.path.exists(rec_dir):
+            raise InvalidRecordingException(
+                reason=f"Target at path does not exist: {rec_dir}",
```

Comment:
What is a target? Maybe replace with "dragged files/folder"?

Reply:
I considered this, but since this class has no knowledge of who and how is passing it the `rec_dir` path, `"dragged"` is too specific and can be misleading too.

`"target"` here pretty much means `"filesystem node"`, whatever it may be (directory, file, link, etc.), since this is what we're testing with `os.path.exists`.

Reply:
Agreed.